mysql-test/suite/innodb/r/foreign_null.result

```diff
@@ -0,0 +1,156 @@
+ALTER TABLE t2 MODIFY COLUMN f1 INT NOT NULL;
+ERROR HY000: Column 'f1' cannot be NOT NULL: needed in a foreign key constraint 't2_ibfk_1' SET NULL
+DROP TABLE t2, t1;
+# modify child column NOT NULL ON DELETE CASCADE..parent column NULL
+CREATE TABLE t1(f1 INT, f2 INT, PRIMARY KEY(f1), KEY(f2))ENGINE=InnoDB;
+CREATE TABLE t2(f1 INT, FOREIGN KEY(f1) REFERENCES t1(f2) ON DELETE CASCADE)ENGINE=InnoDB;
+ALTER TABLE t2 MODIFY COLUMN f1 INT NOT NULL;
+DROP TABLE t2, t1;
+# modify child column NOT NULL ON UPDATE SET NULL
+CREATE TABLE t1(f1 INT, f2 INT, PRIMARY KEY(f1), KEY(f2))ENGINE=InnoDB;
+CREATE TABLE t2(f1 INT, f2 INT, FOREIGN KEY(f1) REFERENCES t1(f1) ON UPDATE SET NULL)ENGINE=InnoDB;
+ALTER TABLE t2 MODIFY COLUMN f1 INT NOT NULL;
+ERROR HY000: Column 'f1' cannot be NOT NULL: needed in a foreign key constraint 't2_ibfk_1' SET NULL
+DROP TABLE t2, t1;
+# modify child column NOT NULL ON DELETE SET NULL
+CREATE TABLE t1(f1 INT, f2 INT, PRIMARY KEY(f1), KEY(f2))ENGINE=InnoDB;
+CREATE TABLE t2(f1 INT, f2 INT, FOREIGN KEY (f2) REFERENCES t1(f2) ON DELETE SET NULL)ENGINE=InnoDB;
+ALTER TABLE t2 MODIFY COLUMN f2 INT NOT NULL;
+ERROR HY000: Column 'f2' cannot be NOT NULL: needed in a foreign key constraint 't2_ibfk_1' SET NULL
+DROP TABLE t2, t1;
+# modify child column NOT NULL ON UPDATE RESTRICT..parent column NULL
+CREATE TABLE t1(f1 INT, f2 INT, PRIMARY KEY(f1), KEY(f2))ENGINE=InnoDB;
+CREATE TABLE t2(f1 INT, f2 INT, FOREIGN KEY (f2) REFERENCES t1(f2) ON UPDATE RESTRICT)ENGINE=InnoDB;
+ALTER TABLE t2 MODIFY COLUMN f2 INT NOT NULL;
+DROP TABLE t2, t1;
+# modify child column NOT NULL ON DELETE RESTRICT..parent column NULL
+CREATE TABLE t1(f1 INT, f2 INT, PRIMARY KEY(f1), KEY(f2))ENGINE=InnoDB;
+CREATE TABLE t2(f1 INT, f2 INT, FOREIGN KEY (f2) REFERENCES t1(f2) ON DELETE RESTRICT)ENGINE=InnoDB;
+ALTER TABLE t2 MODIFY COLUMN f2 INT NOT NULL;
+DROP TABLE t2, t1;
+# modify child column NOT NULL ON UPDATE NO ACTION..PARENT COLUMN NULL
+CREATE TABLE t1(f1 INT, f2 INT, PRIMARY KEY(f1), KEY(f2))ENGINE=InnoDB;
+CREATE TABLE t2(f1 INT, f2 INT, FOREIGN KEY (f2) REFERENCES t1(f2) ON UPDATE NO ACTION)ENGINE=InnoDB;
+ALTER TABLE t2 MODIFY COLUMN f2 INT NOT NULL;
+DROP TABLE t2, t1;
+# modify child column NOT NULL ON DELETE NO ACTION..PARENT COLUMN NULL
+CREATE TABLE t1(f1 INT, f2 INT, PRIMARY KEY(f1), KEY(f2))ENGINE=InnoDB;
+CREATE TABLE t2(f1 INT, f2 INT, FOREIGN KEY (f2) REFERENCES t1(f2) ON DELETE NO ACTION)ENGINE=InnoDB;
+ALTER TABLE t2 MODIFY COLUMN f2 INT NOT NULL;
+DROP TABLE t2, t1;
+# modify parent column NULL ON UPDATE CASCADE child column NOT NULL
+CREATE TABLE t1(f1 INT, f2 INT NOT NULL, PRIMARY KEY(f1), KEY(f2))ENGINE=InnoDB;
+CREATE TABLE t2(f1 INT NOT NULL, FOREIGN KEY(f1) REFERENCES t1(f2) ON UPDATE CASCADE)ENGINE=InnoDB;
+ALTER TABLE t1 MODIFY COLUMN f2 INT;
+ERROR HY000: Cannot change column 'f2': used in a foreign key constraint 't2_ibfk_1' of table 'test.t2'
```

Comment:
I don’t see an immediate problem if the child column allows `NULL` values and the parent column is `NOT NULL`. Yes, the foreign key constraint would prevent a `NULL` from being inserted into the child table.

In fact I think that that this needs to be allowed. Someone could want to change the column from `NOT NULL` to `NULL` in both the child and the parent table. The logical way to do that is to first alter the child table and then the parent table.

Reply:
Sorry, I had overlooked the `ON UPDATE CASCADE`. This one is definitely behaving correctly.

If there is not yet this kind of a case that uses something else (say, `ON DELETE CASCADE` or no `ON` clause at all), please add those variants.

Reply:
This variant is already added as a part of the test case.